compiler/rustc_target/src/target_features.rs

```diff
@@ -316,7 +316,7 @@ const X86_ALLOWED_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
     ("lahfsahf", Unstable(sym::lahfsahf_target_feature), &[]),
     ("lzcnt", Stable, &[]),
     ("movbe", Stable, &[]),
-    ("pclmulqdq", Stable, &[]),
+    ("pclmulqdq", Stable, &["sse2"]),
```

Comment:
Based on looking at other impls, e.g. C#'s version of this https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.x86.pclmulqdq.x64?view=net-8.0, the change looks correct to me.